### ISSUE TITLE
yoda_condition false positive for bit field test

### ISSUE THREAD

**Issue by Bouke** [realm/SwiftLint#4081]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

When testing an integer bit field the yoda_condition rule triggers a false positive.

```swift
if flags & 1 == 1 {
    // ...
}

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
file.swift:2:1: warning: Yoda condition rule Violation: The variable should be placed on the left, the constant on the right of a comparison operator. (yoda_condition)
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.47.1
* Installation method used (Homebrew, CocoaPods, building from source, etc)? brew
* Paste your configuration file: N/A
* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? N/A
* Which Xcode version are you using (check `xcodebuild -version`)? N/A
* Do you have a sample that shows the issue?

```bash
> echo "if b & 1 == 1" | swiftlint lint --no-cache --use-stdin --enable-all-rules
<nopath>:1:1: warning: Yoda condition rule Violation: The variable should be placed on the left, the constant on the right of a comparison operator. (yoda_condition)
Done linting! Found 1 violation, 0 serious in 1 file.
```
